### PR TITLE
What the government doesn't want you to know about visualization

### PR DESCRIPTION
**Pull request by KonradBreitsprecher**
Description of changes:
 - Added documentation for the mayavi and openGL visualizer.
- Corrected/Tested visualization scripts
- Minor fixes for openGL visualizer

PR Checklist
------------
 - [♥] Tests?
   - [♥] Interface
   - [♥] Core 
 - [♥] Docs?

### CODE REVIEW

File: doc/ug/io.tex [hunk 1]
```diff
@@ -1217,6 +1217,256 @@ \section{Error handling}
...
+\label{sec:visualizer}
+\index{Visualization}
+
+With the python interface, \es features two possibilities for
+online-visualization: 
+
+\begin{enumerate}
+\item Using the mlab module to drive \emph{Mayavi, a "3D scientific data
+visualization and plotting in Python"}. Mayavi has a user-friendly GUI to
+specify the appearance of the output. Additional requirements: XXXXXX.
```
Comment (fweik) [inline doc/ug/io.tex]: ?

Comment (KonradBreitsprecher) [inline doc/ug/io.tex]: Left that open for @mkuron but forgot to discuss..
Also just saw that there's a blank page for 'latex' reasons and the pysyntax environment doesn't seem to support argument lists spanning over more than one page. @hmenke?

Comment (mkuron) [inline doc/ug/io.tex]: What requirements do you mean?

File: doc/ug/io.tex [hunk 2]
```diff
@@ -1217,6 +1217,256 @@ \section{Error handling}
...
+setup and equilibration process.
+
+\subsection{General usage}
+
+The recommended usage of both tools is similar: Create the visualizer of your
+choice and pass it the \lstinline{espressomd.System()} object. Then write your
+integration loop in a seperate function, which is started in a non-blocking
+thread. Whenever needed, call \lstinline{update()} to synchronize the renderer
+with your system. Finally start the blocking visualization window with
+\lstinline{start()}.
```
Comment (mkuron) [inline doc/ug/io.tex]: This needs example code.

File: doc/ug/io.tex [hunk 3]
```diff
@@ -1217,6 +1217,256 @@ \section{Error handling}
...
+have suitable default values for most simulations. Colors for particles, bonds
+and constraints are specified by RGBA arrays, materials by an array for the
+ambient, diffuse and specular (ADS) components. To distinguish particle
+groups, arrays of RGBA or ADS entries are used, which are indexed circularly by
+the numerical particle type.
+
+\subsubsection{Keyboard controls}
+
+The camera is controlled via mouse (camera direction), WASD-Keyboard control
+(camera movement) and the key pairs QE, RF, ZC (camera roll).
```
Comment (mkuron) [inline doc/ug/io.tex]: Why QE, RF, ZC? Better use something that's together pairwise, like IJ, OK, PL. Also, how does WASD work in three dimensions?

File: doc/ug/io.tex [hunk 4]
```diff
@@ -1217,6 +1217,256 @@ \section{Error handling}
...
+\item samples/python/visualization.py: LJ-Liquid with live plotting.
+\item samples/python/visualization_bonded.py: Sample for bond visualization.
+\item samples/python/billard.py: Simple billard game including many features of the openGL visualizer.
+\item samples/python/visualization_openGL.py: Timer and keyboard callbacks for the openGL visualizer.
+\item doc/tutorials/python/02-charged_system/scripts/nacl_units_vis.py: Periodic NaCl crystal, see tutorial "Charged Systems".
+\item doc/tutorials/python/02-charged_system/scripts/nacl_units_confined_vis.py: Confined NaCl with interactively adjustable electric field, see tutorial "Charged Systems".
+\item doc/tutorials/python/08-visualization/scripts/visualization.py: LJ-Liquid visualization along with tutorial "Visualization".
+\end{itemize}
+
+Finally, it is recommended to go through tutorial "02-Charged Systems" and "08-Visualization" for further code explanations and examples.
```
Comment (mkuron) [inline doc/ug/io.tex]: Why 02-Charged-Systems?

File: samples/python/visualization_bonded.py [hunk 5]
```diff
@@ -8,41 +8,37 @@
...
+visualizer = visualization.openGLLive(system, bond_type_radius= [0.3])
+
+system.minimize_energy.init(f_max = 10, gamma = 50.0, max_steps = 1000, max_displacement= 0.2)
+system.minimize_energy.minimize()
 
 def main():
-    for i in range(0,int_n_times):
-        print(i)
-        integrate.integrate(int_steps)
+    while True:
```
Comment (mkuron) [inline samples/python/visualization_bonded.py]: You have removed the use of `int_n_times` and `int_steps`, so you should also remove their definitions.

File: samples/python/visualization_openGL.py [hunk 6]
```diff
@@ -15,7 +15,7 @@ def increaseTemp():
         global temperature
         temperature += 0.1
         system.thermostat.set_langevin(kT=temperature, gamma=1.0)
-        print system.temperature 
+        print temperature
```
Comment (mkuron) [inline samples/python/visualization_openGL.py]: `system.temperature` contains the current temperature, while `temperature` is the target temperature. I think it makes more sense to print the current one.

File: src/python/espressomd/visualizationOpenGL.pyx [hunk 7]
```diff
@@ -21,13 +21,13 @@ class openGLLive:
             'window_size':				  [800, 800],
             'name':						  'Espresso Visualization',
             'background_color':			  [0, 0, 0],
-            'update_fps':				  30,
+            'update_fps':				  30.0,
```
Comment (mkuron) [inline src/python/espressomd/visualizationOpenGL.pyx]: Integer values should also be accepted, just convert them to float internally.

File: doc/ug/io.tex (hunk 1 above)
Comment (hmenke) [inline doc/ug/io.tex]: Fixed.

**Comment by mkuron** [comment]
#988 contains the import changes you requested.

File: doc/ug/io.tex (hunk 2 above)
Comment (KonradBreitsprecher) [inline doc/ug/io.tex]: Done. Mayavi has problems if there is only a single particle, so the minimal code now has two particles..

File: doc/ug/io.tex (hunk 3 above)
Comment (KonradBreitsprecher) [inline doc/ug/io.tex]: QE RF ZC are around the WASD keys, which is FPS-like control.

File: doc/ug/io.tex (hunk 4 above)
Comment (KonradBreitsprecher) [inline doc/ug/io.tex]: 02 has visualization examples

File: samples/python/visualization_bonded.py (hunk 5 above)
Comment (KonradBreitsprecher) [inline samples/python/visualization_bonded.py]: Did this I think...

File: samples/python/visualization_openGL.py (hunk 6 above)
Comment (KonradBreitsprecher) [inline samples/python/visualization_openGL.py]: `system.temperature` is depricated, 
seems like the way of choice now is 
`system.thermostat.get_state()[0]['kT']`

File: src/python/espressomd/visualizationOpenGL.pyx (hunk 7 above)
Comment (KonradBreitsprecher) [inline src/python/espressomd/visualizationOpenGL.pyx]: Done.

**Pull request merged**